src/parser.cpp

```diff
@@ -1201,7 +1201,7 @@ const Statement *Parser::parseCaseStatement()
         }
         ++i;
         std::vector<const Statement *> statements;
-        while (tokens[i].type != WHEN && tokens[i].type != ELSE && tokens[i].type != END && tokens[i].type != END_OF_FILE) {
+        while (tokens[i].type != WHEN && tokens[i].type != END && tokens[i].type != END_OF_FILE && tokens[i+1].type != OTHERS) {
```

Comment:
Why are you checking for `OTHERS` here? That isn't necessary. A when clause ends with another `WHEN`, or an `END` (no longer with an `ELSE`, so all you should need to do is remove that).